src/extra/libs/rlottie/rlottiedec.c

```diff
@@ -0,0 +1,481 @@
+/**
+ * @file rlottiedec.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+#include <string.h>
+
+#if LV_USE_RLOTTIE
+#include "rlottiedec.h"
+#include <rlottie_capi.h>
+
+
+#ifndef RLOTTIE_FEATURE_PARTIAL_RENDER
+    #pragma message "Your version of Rlottie does not support partial decoding. Most of the code below " \
+    "will fail to work efficiently without this support."
+    #define FORCE_ALLOCATING_WHOLE_PICTURE 1
+#else
+    #define FORCE_ALLOCATING_WHOLE_PICTURE 0
+#endif
+
+
+#ifndef RLOTTIE_FEATURE_RO_JSON
+    #pragma message "Your version of Rlottie does not support partial read-only JSON parsing, a copy is required"
+    #define FORCE_JSON_COPY 1
+#else
+    #define FORCE_JSON_COPY 0
+#endif
+
+#if FORCE_JSON_COPY == 1 || FORCE_ALLOCATING_WHOLE_PICTURE == 1
+    #pragma message "Your version of Rlottie does not support last frame decoding"
+    #pragma message "You should be using https://github.com/lvgl/rlottie version instead"
+#endif
+
+
+/*********************
+ *      DEFINES
+ *********************/
+#define LV_ARGB32   32
+#define RLOTTIE_MAX_BUFSIZE     4096
+
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+typedef struct {
+    uint32_t * allocated_buf;
+    lv_coord_t lines_in_buf;
+    size_t scanline_width;
+    lv_coord_t top;
+    size_t last_rendered_frame;
+} lv_rlottie_dec_context_t;
+
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static lv_res_t decoder_accept(const lv_img_src_t * src, uint8_t * caps);
+static lv_res_t decoder_open(lv_img_dec_dsc_t * dsc, const lv_img_dec_flags_t flags);
+
+
+static lv_res_t decoder_read_line(lv_img_dec_dsc_t * dsc,
+                                  lv_coord_t x, lv_coord_t y, lv_coord_t len, uint8_t * buf);
+
+static void decoder_close(lv_img_dec_dsc_t * dsc);
+
+static void set_caps(uint8_t * caps);
+static lv_res_t init_dec_ctx(rlottiedec_ctx_t * dec_ctx);
+static lv_res_t render_animation(lv_rlottie_dec_context_t * context_ctx, rlottiedec_ctx_t * dec_ctx, lv_coord_t w,
+                                 lv_coord_t h);
+static void set_desc_size(lv_img_dec_dsc_t * dsc, size_t w, size_t h);
+
+
+#if FORCE_JSON_COPY
+Lottie_Animation * lottie_animation_from_rodata(const char * src, const size_t len, const char * dirname)
+{
+    char * json = (char *)lv_mem_alloc(len + 1);
+    LV_ASSERT_MALLOC(json);
+    if(!json) return NULL;
+
+    lv_memcpy(json, src, len);
+    json[len] = 0;
+
+    Lottie_Animation * animation = lottie_animation_from_data(json, NULL, dirname);
+    lv_mem_free(json);
+    return animation;
+}
+#endif
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static int rlottiedec_init = 0;
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+void lv_rlottie_init(void)
```

Comment:
The file also should be called `lv_rlottie.c`.

Reply:
You mean, merge both C files ?